File: backend/internal/entity/provider.go
package entity

import (
	"time"

	"gorm.io/datatypes"
)

type Provider struct {
	ID              int64          `gorm:"primaryKey" json:"id"`
	Name            string         `json:"name"`
	Slug            string         `json:"slug"`
	Vendor          string         `json:"vendor"`
	AdapterType     string         `json:"adapter_type"`
	AuthType        string         `json:"auth_type"`
	BaseURL         string         `json:"base_url"`
	APIKeyEncrypted string         `json:"-"`
	ConfigJSON      datatypes.JSON `gorm:"column:config_json" json:"config_json"`
	Status          string         `json:"status"`
	Description     string         `json:"description"`
	CreatedAt       time.Time      `json:"created_at"`
	UpdatedAt       time.Time      `json:"updated_at"`
}

File: frontend/src/types/provider.ts
export interface Provider {
  id: number;
  name: string;
  slug: string;
  vendor: "openai" | "anthropic" | "google" | "custom";
  adapter_type: "openai_compatible" | "anthropic" | "claude_code" | "gemini" | "vertexai";
  auth_type: "api_key" | "local_oauth" | "adc" | "claude_oauth" | "none";
  base_url: string;
  config_json: Record<string, unknown>;
  status: "active" | "disabled";
  description: string;
  created_at: string;
  updated_at: string;
}

export interface ProviderInput {
  name: string;
  slug: string;
  vendor: Provider["vendor"];
  adapter_type: Provider["adapter_type"];
  auth_type: Provider["auth_type"];
  base_url: string;
  api_key_encrypted?: string;
  config_json: Record<string, unknown>;
  status: Provider["status"];
  description: string;
}
